src/index.ts

```diff
@@ -2,27 +2,33 @@ import * as engine from './engine';
 import { getPageFeatures } from './features';
 import { viewStore, matchedAudienceStore } from './store';
 import { timeStampInSecs } from './utils';
-import { hasGdprConsent } from './gdpr';
+import { waitOnConsent } from './gdpr';
 import {
   PageFeatureGetter,
   MatchedAudience,
   AudienceDefinition,
 } from '../types';
 
+export { runOnConsent, waitOnConsent } from './gdpr';
+
 interface Config {
   pageFeatureGetters: PageFeatureGetter[];
   audienceDefinitions: AudienceDefinition[];
   vendorIds?: number[];
   omitGdprConsent?: boolean;
+  allowMultipleRuns?: boolean; // For testing purposes only
 }
 
+let hasRun = false;
```

Comment:
Again question on why this is needed?

Reply:
I was thinking about how to set up the logic with the event listener, but ended up going with `waitOnConsent` instead. This was just leftover as a todo for cleanup after finalizing the design

Reply:
oh ok cool!